hazelcast/src/main/java/com/hazelcast/sql/impl/state/QueryClientStateRegistry.java

```diff
@@ -199,36 +207,56 @@ public void closeOnError(QueryId queryId) {
     private void close0(QueryClientState clientCursor) {
         clientCursor.getSqlResult().close();
 
-        deleteClientCursor(clientCursor);
+        deleteClientCursor(clientCursor.getQueryId());
     }
 
     public void shutdown() {
         clientCursors.clear();
     }
 
     public void update(Set<UUID> activeClientIds) {
+        long currentTime = System.currentTimeMillis();
```

Comment:
We should use `nanoTime` for elapsed time.

Reply:
What for? Both `currentTimeMillis` and `nanoTime` is used in the codebase, with more usages of the `currentTimeMillis`. We are not doing any precise time tracking here, even errors ~1s are ok for us because the goal is to clean some rare reordered messages eventually.

Reply:
`currentTimeMillis` isn't monotonic, it can jump by any amount and in any direction. Of course in most cases it doesn't, but it's just not designed to measure elapsed time. It's not only about resolution.

Reply:
Makes sense, fixed.